pandas/core/dtypes/astype.py

```diff
@@ -246,3 +247,55 @@ def astype_array_safe(
             raise
 
     return new_values
+
+
+def astype_is_view(dtype: DtypeObj, new_dtype: DtypeObj) -> bool:
+    """Checks if astype avoided copying the data.
+
+    Parameters
+    ----------
+    dtype : Original dtype
+    new_dtype : target dtype
+
+    Returns
+    -------
+    True if new data is a view or not guaranteed to be a copy, False otherwise
+    """
+    if dtype == new_dtype:
+        return True
+
+    elif isinstance(dtype, np.dtype) and isinstance(new_dtype, np.dtype):
+        # Only equal numpy dtypes avoid a copy
+        return False
+
+    elif is_string_dtype(dtype) and is_string_dtype(new_dtype):
+        # Potentially! a view when converting from object to string
+        return True
+
+    elif is_object_dtype(dtype) and new_dtype.kind == "O":
+        # When the underlying array has dtype object, we don't have to make a copy
+        return True
+
+    elif dtype.kind in "mM" and new_dtype.kind in "mM":
+        dtype = getattr(dtype, "numpy_dtype", dtype)
```

Comment:
@phofl i finally got around to seeing this.  this numpy_dtype getattr check looks like it is for arrow/masked arrays.  can we avoid this pattern?  it makes it difficult to tell where we are special-casing

Reply:
id prefer for these to be as explicit as possible.  i suspect this is intended for something more specific than just ExtensionDtype

Reply:
Not really, basically all extension dtypes that have a numpy dtype

Reply:
They could yes, but right now this is implemented defensively to track more references than necessary if we are not able to determine with certainty that a copy happened.